slides.md

```diff
@@ -156,57 +156,97 @@ import anotherFunction, {aFunction} from "./someFile";
 
 ---
 # Hello Todo!
-1. Edit `App.js` so that the list rendered from an array:
+Currently `App.js` is quite boring, it would be great if instead of having the todos hardcoded, we could somehow cycle through an array of todos and show a `Text` per element.
+
+First of all, we need to extract the data for the todos, in `App.js`:
 ```javascript
 const todos = [{
     name: "Take the dog out",
-  }, ...
+  }, // ...
 ];
 ```
-You can use `todos.map()` to return a component, return a `Text` component showing `todo.name`.
+
+Inside `render()`, you can then map those todos to a certain element:
+
+```javascript
+const todoElements = todos.map(todo => <Text>{todo.name}</Text>);
+```
+
+> Inside a component you can include a JS expression with `{expression}`, in this case `{todo.name}`.
+
+Since `todoElements` is an expression, we can include it inside our `View`, instead of the 3 `Text` elements, remember to use `{}`!
 
 ---
 # Keys in loops
-Components rendered with loops require a `key` property that identifies them.
-Keys are important for react to be able to understand which views have been added, removed or altered. Use `todo.name` as a key for now.
 
----
-# Conventions
-- Sometimes I'll reference React Native components, such as `Text`. These can be imported with `import { ComponentName } from "react-native";`.
-- When you need to create a new component, save it in `ComponentName.js` and import it with `import { ComponentName } from "./ComponentName';`.
+If you reloaded, you'll notice there's a small yellow warning on the bottom of the app. Components rendered with loops require a `key` property that identifies them.
+Keys are important for react to be able to understand which views have been added, removed or altered.
+
+> As you can do inside a component, you can use `propName={expression}` to pass properties. If it's a hardcoded string, prefer `propName="value"`.
+
+Before moving to the next step, let's use `todo.name` as a key for our elements.
 
 ---
 # Abstracting away
-The app looks amazing! But it would be great to have a component `Todo` instead of using `Text`. The simplest components can be functions that accept properties and return other components, strings or `null`. Pay attention to the brackets for the function, we'll go over that next.
+
+Now that our single todo is well isolated, we can extract it to a separate component.
+
+> The simplest component is a function that returns a tree of elements. It must have no side effects and return the same result given the same input.
```

Comment:
if this is the definition of something, write that something 😄 
i.e. 

Stateless component:
> The simplest...

Reply:
I think  I went a bit overboard with the block quotes, will change :P

Reply:
20324b1